services/collaboration/pkg/middleware/wopicontext.go

```diff
@@ -120,12 +123,24 @@ func WopiContextAuthMiddleware(cfg *config.Config, st microstore.Store, next htt
 			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 			return
 		}
-		user, _, err := tokenManager.DismantleToken(ctx, wopiContextAccessToken)
+		user, scope, err := tokenManager.DismantleToken(ctx, wopiContextAccessToken)
 		if err != nil {
 			wopiLogger.Error().Err(err).Msg("failed to dismantle reva token manager")
 			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 			return
 		}
+
+		for k, v := range scope {
+			if strings.HasPrefix(k, "publicshare:") && v.Resource.Decoder == "json" {
+				share := &link.PublicShare{}
+				err := utils.UnmarshalJSONToProtoV1(v.Resource.Value, share)
+				if err == nil {
+					claims.WopiContext.publicShare = share
+					break
+				}
+			}
+		}
```

Comment:
This might need some changes...
Do we need some debug log to know what public shares are we interested in, or how many scopes are detected? In any case, if there is some error unmarshaling the JSON, not only it will be skipped (which might be fine), but we don't know what happen (which is bad). The error needs to be logged at the very least.

Reply:
added the error logging